v3/newrelic/transaction.go

```diff
@@ -269,6 +271,76 @@ func (txn *Transaction) AcceptDistributedTraceHeaders(t TransportType, hdrs http
 	txn.thread.logAPIError(txn.thread.AcceptDistributedTraceHeaders(t, hdrs), "accept trace payload", nil)
 }
 
+//
+// AcceptDistributedTraceHeadersFromJson() works just like AcceptDistributedTraceHeaders(), except
+// that it takes the header data as a JSON string à la DistributedTraceHeadersFromJson(). Additionally
+// (unlike AcceptDistributedTraceHeaders()) it returns an error if it was unable to successfully
+// convert the JSON string to http headers. There is no guarantee that the header data found in JSON
+// is correct beyond conforming to the expected types and syntax.
+//
+func (txn *Transaction) AcceptDistributedTraceHeadersFromJson(t TransportType, jsondata string) error {
+	hdrs, err := DistributedTraceHeadersFromJson(jsondata)
+	if err != nil {
+		return err
+	}
+	txn.AcceptDistributedTraceHeaders(t, hdrs)
+	return nil
+}
+
+//
+// DistributedTraceHeadersFromJson() takes a set of distributed trace headers as a JSON-encoded string
+// and emits a http.Header value suitable for passing on to the
+// txn.AcceptDistributedTraceHeaders() function.
+//
+// For example, given the input string
+//   `{"traceparent": "frob", "tracestate": "blorfl", "newrelic": "xyzzy"}`
+// This will emit an http.Header value with headers "traceparent", "tracestate", and "newrelic".
+//
+// This is a convenience function provided for cases where you receive the trace header data
+// already as a JSON string and want to avoid manually converting that to an http.Header.
```

Comment:
I wonder if we want to add that this is likely when working with trace headers with different language agents? Maybe that would be too wordy. I just want to make sure somebody like Jo Ann would see this and use it.

Reply:
Not a bad idea. I don't mind leaning to more verbosity when it comes to documentation. You never know when that extra sentence or paragraph was just what someone was looking for.